spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpSessionFactoryTests.java

```diff
@@ -218,4 +220,30 @@ void customTimeoutIsApplied() throws IOException {
 		}
 	}
 
+	@Test
+	void clientSessionIsClosedOnSessionClose() throws IOException {
```

Comment:
This test proves nothing.
Please, consider to revise it when you agree with my proposal.

Thanks

Reply:
For the simple use cases for example connecting to a SFTP server every five hours to retrieve files, I don't feel the ```ClientSession``` should remain open after the after the files have been synchronized locally. This fix was an attempt to solve those simple use cases.

How would you recommend handling the simpler use cases?

Reply:
After more thinking, I'll accept your suggestion.
Looks like it works well in combination of `DefaultSftpSessionFactory` and `CachingSessionFactory`.
So, if our session is not valid, it will close all its relevant resources.
The next new session will come with new resources.
If it is valid, then it is just going to be cached with all of its relevant resources.

Confirm, please, that it is what you are intended for.

However I still believe that `isInnerClient` has to be closed in the `destroy()` of the `DefaultSftpSessionFactory`.
Would you mind addressing that as a part of your fix?

Reply:
Yes, that's the intent. 

I'll can add stop method call in ```DisposableBean.destory()``` implementation as well.

Thanks

Reply:
Adding the ```DisposableBean``` interface to DefaultSftpSessionFactory adds the requirement that objects of this type need to be created be created by the BeanFactory. This is not the case in the test suite and cannot be assumed to be the case in existing applications.

Should I proceed with adding the interface?

Thanks

Reply:
The assumption that components from the framework might be used outside of `BeanFactory` is error-prone.
When we talk about Spring, the component must be a bean for the proper dependency injection lifecycle.
So, yes, please, implement that interface and fix respective tests to call `destroy()` directly, if that unit test is not based on `BeanFactory`.